osu.Framework/Input/IHandleNonPositionalInput.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Framework.Input
+{
+    /// <summary>
+    /// Marker interface for interfaces which depend on non-positional input.
+    /// All drawables which implement this interface will have <see cref="Graphics.Drawable.HandleNonPositionalInput"/> set to <see langword="true"/> by default.
+    /// </summary>
+    public interface IHandleNonPositionalInput
```

Comment:
```suggestion
    /// <summary>
    /// Marker interface for drawables which depend on non-positional input.
    /// All drawables which implement this interface will have <see cref="Graphics.Drawable.HandleNonPositionalInput"/> set to <see langword="true"/> by default.
    /// </summary>
    public interface IHandleNonPositionalInput : IDrawable
```
(same applies to other interface)

Reply:
The interface is specifically meant to be inherited directly by interfaces and not drawables, which is why the docs say "interfaces". `IHandleGlobalKeyboardInput` does not force the implementer to be an `IDrawable`, so I dont think this one should either.

Reply:
I'm not sure about that, whichever implements `IHandleNonPositionalInput`/`IHandlePositionalInput` must surely be a drawable type, otherwise the interface is pointless for that type. Also, I can imagine drawables implementing that interface directly, e.g. abstract classes with certain logic that requires input.

Reply:
If so, then the global keyboard interface should have it added too.

Such drawables surely could just override Handles(Non)PositionalInput themselves, or implement an interfce/override a handler method though, no? I imagine the whole class which assigns default values to whether a drawable type does or doesnt handle a given input type is more of a confusion avioder/optimisation rather than *the* way to define it

Reply:
> If so, then the global keyboard interface should have it added too.

I see no problem with doing so? `IRequireHighFrequencyMousePosition` already implements `IDrawable` for what it's worth.

Reply:
> I see no problem with doing so?

Oh, that wasn't a counterpoint, I was just saying that should this interface have it added, I should add it to the global keyboard one too.

Reply:
To reiterate: I'm suggesting for this interface to only ever be inherited by drawables or interfaces that only drawables can inherit, because it's the only place where this interface can function.

If this is focusing on the xmldoc part of the suggestion, then it can be reworded in some form. Having it as `Marker interface for interfaces` definitely reads weird, would suggest `Marker interface for <see cref="IDrawable"/>-based interfaces` at the very least.